README.md

```diff
@@ -12,25 +12,107 @@ gem 'makwa'
 
 And then execute:
 
-    $ bundle install
+```shell
+$ bundle install
+```
 
 Or install it yourself as:
 
-    $ gem install makwa
+```shell
+$ gem install makwa
+```
+
+Makwa extends on the Active Interactions gem, if you do not have this installed, you can add it to your Gemfile:
+
+```ruby
+gem 'active_interaction', '~> 4.1'
+```
+
+or install it manually
+
+```shell
+$ gem install active_interaction --version '~> 4.1' 
+```
 
 ## Usage
 
-TODO: Write usage instructions here
+### Here is an example of an Interaction and a Controller which invoked the Interaction
+
+```ruby
+
+class Create < ApplicationInteraction
+  
+  # Input filters
+  string :email
+  string :name
+  string :password
+  
+  # ActiveModel validations
+  validates :email, presence: true
+  validates :name, presence: true
+  validates :password, presence: true, length: { minimum: 6 }
+  
+  # @return [User]
+  def execute
+	user = User.new(inputs)
+	errors.merge!(user.errors) unless user.update(inputs.except(:user))
+	halt_if_errors!
+	user.update(inputs.except(:user))
+	user
+  end
+  
+end
+```
+
+The interaction file is located at app/interactions/public/users/create.rb. Our documentation contains good tips for naming conventions.
+
+We first specify the input filters. They will check for the presence and correct type of each input argument. They can do type casting if needed, and provide default values. Here we expect three string arguments for email, name, and password.
+
+Next you see input data validations using standard ActiveModel validations. These will be applied to the input arguments. Execution will stop if there are any validation errors. Finally we have the execute method where we implement the behavior. This method is called only if all inputs are present and valid. In the execute method we have access to the errors collection. 
+
+We halt execution if there are any errors. Only if everything has worked as expected so far, do we cause the additional side effects of sending emails, etc. 
+
+Only then do we return the newly created user.
+
+```ruby
+module Public
```

Comment:
Jenn's RubyMine shortcut: `cmd`+`A` to select all, then `cmd`+`option`+`l` to auto-fix the spacing in the whole file